casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -70,6 +72,13 @@ object DeployRuntime {
       }.map(kp("")).value
     }
 
+  def findDeploy[F[_]: Functor: Sync: Time: DeployService](
+      deployId: String
+  ): F[Unit] =
+    gracefulExit(
+      DeployService[F].findDeploy(FindDeployQuery(Base16.unsafeDecode(deployId).toByteString))
```

Comment:
I'm not sure it's safe to do Base16.unsafeDecode here. 

Who calls this function? Is it possible that the deployId is invalid Base16?

Maybe you should move the decoding to DeployService.findDeploy and deal with the error there?

Reply:
Ok, I'll add validation here